spring-javaformat-eclipse/io.spring.javaformat.eclipse/src/io/spring/javaformat/eclipse/gradle/GradleProjectSettingsConfigurator.java

```diff
@@ -16,117 +16,80 @@
 
 package io.spring.javaformat.eclipse.gradle;
 
-import java.io.File;
-import java.io.IOException;
-import java.util.Collection;
-import java.util.LinkedHashSet;
-import java.util.Optional;
-import java.util.Set;
+import java.util.List;
+import java.util.Map.Entry;
 
-import org.eclipse.buildship.core.GradleBuild;
-import org.eclipse.buildship.core.InitializationContext;
 import org.eclipse.buildship.core.ProjectConfigurator;
-import org.eclipse.buildship.core.ProjectContext;
-import org.eclipse.buildship.core.internal.CorePlugin;
-import org.eclipse.buildship.core.internal.workspace.FetchStrategy;
-import org.eclipse.buildship.core.internal.workspace.InternalGradleBuild;
-import org.eclipse.buildship.core.internal.workspace.InternalGradleWorkspace;
-import org.eclipse.buildship.core.internal.workspace.ModelProvider;
-import org.eclipse.core.resources.IProject;
 import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.IProgressMonitor;
-import org.gradle.tooling.CancellationTokenSource;
-import org.gradle.tooling.GradleConnector;
-import org.gradle.tooling.model.GradleTask;
-import org.gradle.tooling.model.eclipse.EclipseProject;
+import org.eclipse.m2e.core.lifecyclemapping.model.IPluginExecutionMetadata;
```

Comment:
I'm surprised to see m2e types being imported into the Gradle settings configurator. Doesn't this need to hook into Buildship somehow?

Reply:
Sorry, you are right. I've repaired it. 
Thanks.